pkg/controller/replication/replication_controller.go

```diff
@@ -478,10 +573,19 @@ func (rm *ReplicationManager) manageReplicas(filteredPods []*api.Pod, rc *api.Re
 		var wg sync.WaitGroup
 		wg.Add(diff)
 		glog.V(2).Infof("Too few %q/%q replicas, need %d, creating %d", rc.Namespace, rc.Name, rc.Spec.Replicas, diff)
+		// needs to set the gvk to generate ownerReference
+		oldKind := rc.Kind
+		oldAPIVersion := rc.APIVersion
+		defer func() {
+			rc.Kind = oldKind
+			rc.APIVersion = oldAPIVersion
+		}()
+		rc.APIVersion = v1.SchemeGroupVersion.String()
+		rc.Kind = "ReplicationController"
```

Comment:
are you writing to an rc that you pull out of the store? generally not a good idea since we store pointers in the store (if another thread checks the rc from the store between your mutations we're toast, and this is likely to happen with SI). Even if you are resetting it in the defer func. Please add a comment about what you're trying to achieve. Suggest doing it without flipping fields back and forth.